mountpoint-s3/src/inode.rs

```diff
@@ -2522,11 +2527,25 @@ mod tests {
         let stat = inode.get_inode_state().unwrap().stat.clone();
         assert!(!stat.is_valid());
 
-        // Should get an error back when calling setattr
+        // Should be able to reset expiry back and make stat valid when calling setattr
         let atime = OffsetDateTime::UNIX_EPOCH + Duration::days(90);
         let mtime = OffsetDateTime::UNIX_EPOCH + Duration::days(60);
-        let result = superblock.setattr(&client, ino, Some(atime), Some(mtime)).await;
-        assert!(matches!(result, Err(InodeError::SetAttrOnExpiredStat(_))));
+        let lookup = superblock
```

Comment:
I still need to see if there is any other possible scenario is where InodeStat may be invalid and does resetting the InodeStat is correct in that case.